src/plugins/mf_priority.cpp

```diff
@@ -0,0 +1,271 @@
+                        void *data)
+{
+    int urgency, userid;
+    char *bank = NULL;
+    int64_t priority;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:i, s{s{s{s?s}}}}",
+                                "urgency", &urgency,
+                                "userid", &userid,
+                                "jobspec", "attributes", "system",
+                                "bank", &bank) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    priority = priority_calculation (p, args, userid, bank, urgency);
+
+    if (flux_plugin_arg_pack (args,
+                              FLUX_PLUGIN_ARG_OUT,
+                              "{s:I}",
+                              "priority",
+                              priority) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_pack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+    return 0;
+}
+
+
+/*
+ * Look up the userid of the submitted job in the multimap; if user is not found
+ * in the map, reject the job saying the user wasn't found in the
+ * flux-accounting database.
+ */
+static int validate_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int userid;
+    char *bank = NULL;
+    std::map<int, std::tuple<std::string, double>>::iterator it;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s{s{s{s?s}}}}",
+                                "userid", &userid,
+                                "jobspec", "attributes", "system",
+                                "bank", &bank) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    // searching element in std::map by key
+    it = users.find (userid);
+    if (it == users.end ())
+        return flux_jobtap_reject_job (p, args,
+                                       "user not found in flux-accounting DB");
+
+    // reject job if user belongs to two or more banks and does not specify
+    // bank on job submission
+    if (users.count (userid) > 1 && bank == NULL)
+        return flux_jobtap_reject_job (p, args,
+                                       "user belongs to two or more banks; " \
+                                       "please specify which bank to charge");
+
+    return 0;
+}
+
+
+char v[] = "job.validate";
+char p[] = "job.state.priority";
+char g[] = "job.priority.get";
```

Comment:
Thanks for pointing this out - the reason I defined them is because C++ gave me a warning about converting strings to `char *` if I just use the strings directly in `tab[]`:

```console
warning: ISO C++ forbids converting a string constant to ‘char*’
```

Is it OK to ignore this warning?

Reply:
Ah, this is because the topic string in `struct flux_plugin_handler` is `char *topic` instead of `const char *topic`. This is a bug in the `flux_plugin_t` interface. I'll open a bug in flux-core and fix it, and then those warnings should be resolved.

Sorry about that!

Reply:
No worries! Thanks for helping me walk through that. 🙂

Reply:
A fix is up for this at flux-framework/flux-core#3720